packages/cafs/src/index.ts

```diff
@@ -0,0 +1,81 @@
+import getStream = require('get-stream')
+import path = require('path')
+import exists = require('path-exists')
+import pathTemp = require('path-temp')
+import renameOverwrite = require('rename-overwrite')
+import ssri = require('ssri')
+import { Hash } from 'ssri'
+import addFilesFromDir from './addFilesFromDir'
+import addFilesFromTarball from './addFilesFromTarball'
+import checkFilesIntegrity from './checkFilesIntegrity'
+import writeFile from './writeFile'
+
+export { checkFilesIntegrity }
+
+export default function createCafs (cafsDir: string, ignore?: ((filename: string) => Boolean)) {
```

Comment:
you mean, the ignore function? You are right. I don't know what would be the expected result here. Maybe it should fail. I am not sure.

Reply:
In any case, even if we allow it to fail, it should fail with a meaningful error but I created a separate issue for this because it already works like this.

https://github.com/pnpm/pnpm/issues/2500

Reply:
Any async function **must** be wrapped in try/catch (or be chained with `.catch()`) somewhere in the call chain. Uncaught promise errors currently result in a nasty error message being displayed (shown below). As stated in the message, in the future they will cause the immediate termination of the process, which is not what you want to happen.

(node:41035) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:41035) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

Reply:
Sure, I know about that. And this error will be caught by a try/catch in the `@pnpm/resolve-dependencies` package (which satisfies the condition "somewhere in the call chain")

Reply:
Just wanted to make sure. 👍